### PR TITLE
Add captions to plotted outputs

### PR DESCRIPTION
**Pull request by stefanv**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -87,6 +89,7 @@ $(BUILD_HTMLBOOK)/%.html: $(BUILD_NB)/%.ipynb
 	tools/wrap_callouts.py $@ > $@.tagged && mv $@.tagged $@
 	cp $@ /tmp
 	tools/wrap_figure.py $@ > $@.figures && mv $@.figures $@
+	tools/caption_crunch.py $@ > $@.captions && mv $@.captions $@
```
Comment (jni) [inline Makefile]: Caption crunch!!!! 🤣 🤣 🤣 🤣 👏👏👏👏👏👏👏 🥇

Comment (stefanv) [inline Makefile]: :D

File: markdown/ch2.markdown [hunk 2]
```diff
@@ -68,17 +68,20 @@ def quantile_norm_log(X):
...
+Arrays can be operated on along *axes*. In the first line of `quantile_norm`, we sort the data along each column just by specifying an `axis` parameter to `np.sort`. We then take the mean along each row by specifying a *different* `axis`.
+
+Arrays underpin the scientific Python ecosystem. The `scipy.stats.rankdata` function operates not on Python lists, but on NumPy arrays. This is true of many scientific libraries in Python.
+
+Even functions that don't have an `axis=` keyword can be made to operate along axees by NumPy's `apply_along_axis` function.
+
+Arrays support many kinds of data manipulation through *fancy indexing*:
+`Xn = quantiles[ranks]`. This is possibly the trickiest part of NumPy, but
+also among the most useful. We will explore it further in the text that
+follows.
```
Comment (jni) [inline markdown/ch2.markdown]: There is only a tiny bit of math in this list — maybe the right approach is to remove the math here?

Comment (stefanv) [inline markdown/ch2.markdown]: Probably, I just did something That Worked (TM) ;)

**Comment by stefanv** [comment]
fig2 6

**Comment by stefanv** [comment]
toc

**Comment by jni** [comment]
@stefanv here's the Wercker error:

```
Traceback (most recent call last):
  File "tools/latex_to_mathml.py", line 83, in <module>
    print(mathml_data)
UnicodeEncodeError: 'ascii' codec can't encode character '\u2014' in position 61: ordinal not in range(128)
Traceback (most recent call last):
  File "tools/footnote_fixer.py", line 46, in <module>
    print(re.sub('\[\^(.*?)\][^:]', insert_footnote, data))
UnicodeEncodeError: 'ascii' codec can't encode character '\u2014' in position 61: ordinal not in range(128)
```

The unicode character is an em-dash. I'm not sure where the ASCII is coming from...?

**Comment by stefanv** [comment]
FINALLY

**Pull request merged**